### PR TITLE
Clarify mixing C++ and HIP sources via CMake

### PR DESCRIPTION
**Pull request by MathiasMagnus**
Fixes `swdev-384588`

### CODE REVIEW

**Review by saadrahim — COMMENTED** [review]
@LisaDelaney can you review for language?

File: docs/conceptual/cmake-packages.rst [hunk 1]
```diff
@@ -140,6 +144,11 @@ any C or C++ compiler can be used. The ``find_package(hip)`` provides the
     add_executable(MyApp ...)
     target_link_libraries(MyApp PRIVATE hip::host)
 
+When mixing ``CXX`` sources using the host API only and HIP sources with
```
Comment (saadrahim) [inline docs/conceptual/cmake-packages.rst:147]: @LisaDelaney this sentence needs a second look

File: docs/conceptual/cmake-packages.rst [hunk 2]
```diff
@@ -97,6 +97,10 @@ there's a catch-all, last resort variable consulted locating this file,
 ``-D CMAKE_HIP_COMPILER_ROCM_ROOT:PATH=`` which should be set the root of the
 ROCm installation.
 
+.. note::
+    Imported targets defined by `hip-lang-config.cmake` are for internal use for
```
Comment (LisaDelaney) [inline docs/conceptual/cmake-packages.rst:101]: Can we just say: "Imported targets defined by `hip-lang-config.cmake` are for internal use only."

File: docs/conceptual/cmake-packages.rst (hunk 1 above)
Comment (LisaDelaney) [reply docs/conceptual/cmake-packages.rst:147]: - Do we need to say what "CXX" is in the first instance?
- As for rephrasing  (it's a confusing/dense sentence), how do you feel about this? "When mixing ``CXX`` sources that use 1) host API only and 2) HIP with device code, link only to `hip::host`. If the HIP sources don't have the `.hip` extension, you must use `set_source_files_properties(<hip_sources>... PROPERTIES LANGUAGE HIP)`."

File: .github/workflows/linting.yml
Comment (saadrahim) [inline .github/workflows/linting.yml:1]: Please revert this change and let's get this PR passing.

**Review by saadrahim — CHANGES_REQUESTED** [review]

File: docs/about/compatibility/docker-image-support-matrix.rst [hunk 3]
```diff
@@ -3,7 +3,7 @@ Docker image support matrix
 ******************************************************************
 
 AMD validates and publishes `PyTorch <https://hub.docker.com/r/rocm/pytorch>`_ and
-`TensorFlow <https://hub.docker.com/r/rocm/tensorflow>`_ containers on Docker Hub. The following
+`TensorFlow <https://hub.docker.com/r/rocm/tensorflow>`_ containers on DockerHub. The following
```
Comment (Rmalavally) [inline docs/about/compatibility/docker-image-support-matrix.rst:6]: @MathiasMagnus  Docker Hub is two words, with D and H in uppercase. It's a repo name, so it will be good to follow the convention.

https://docs.docker.com/docker-hub/

@LisaDelaney Please add a guideline.

File: docs/conceptual/cmake-packages.rst (hunk 1 above)
Comment (MathiasMagnus) [reply docs/conceptual/cmake-packages.rst]: Rephrased.

File: docs/conceptual/cmake-packages.rst (hunk 2 above)
Comment (MathiasMagnus) [reply docs/conceptual/cmake-packages.rst]: Simplified.

File: docs/about/compatibility/docker-image-support-matrix.rst (hunk 3 above)
Comment (MathiasMagnus) [reply docs/about/compatibility/docker-image-support-matrix.rst]: This was an erronous attempt at pleasing the spellchecker. The change has been reverted.

File: .github/workflows/linting.yml
Comment (MathiasMagnus) [reply .github/workflows/linting.yml:1]: Reverted.

File: docs/conceptual/cmake-packages.rst [hunk 4]
```diff
@@ -98,6 +98,10 @@ there's a catch-all, last resort variable consulted locating this file,
 ``-D CMAKE_HIP_COMPILER_ROCM_ROOT:PATH=`` which should be set the root of the
 ROCm installation.
 
+.. note::
+    Imported targets defined by `hip-lang-config.cmake` are for internal use
```
Comment (Rmalavally) [inline docs/conceptual/cmake-packages.rst:102]: @MathiasMagnus what is the objective of this note? How would it help external users to know that this is only for internal consumption? How will our users benefit?
